tilequeue/command.py

```diff
@@ -983,27 +1021,42 @@ def tilequeue_prune_tiles_of_interest(cfg, peripherals):
     prune_cfg = cfg.yml.get('toi-prune', {})
 
     redshift_cfg = prune_cfg.get('redshift', {})
-    redshift_uri = redshift_cfg.get('database-uri')
-    assert redshift_uri, ("A redshift connection URI must "
+    db_conn_info = redshift_cfg.get('database-uri') or cfg.postgresql_conn_info
+    assert db_conn_info, ("A redshift connection URI or postgres configuration must "
                           "be present in the config yaml")
+    
+    is_postgres_conn_info = isinstance(db_conn_info, dict)
+    if is_postgres_conn_info:
+        # use first database specified in postgres config for connection
+        dbname = db_conn_info.pop('dbnames')[0]
+        db_conn_info = dict(db_conn_info, dbname=dbname)
 
     redshift_days_to_query = redshift_cfg.get('days')
     assert redshift_days_to_query, ("Number of days to query "
                                     "redshift is not specified")
 
     redshift_zoom_cutoff = int(redshift_cfg.get('max-zoom', '16'))
 
-    s3_parts = prune_cfg.get('s3')
-    assert s3_parts, ("The name of an S3 bucket containing tiles "
-                      "to delete must be specified")
+    # flag indicating that s3 entry in toi-prune is used for s3 store 
+    legacy_fallback = 's3' in prune_cfg
+    store_parts = prune_cfg.get('s3') or prune_cfg.get('store')
+    assert store_parts, ("The configuration of a store containing tiles "
+                      "to delete must be specified under toi-prune:store or toi-prune:s3")
+    # explictly override the store configuration with values provided in toi-prune:s3
+    if legacy_fallback:
+        cfg.store_type = 's3'
+        cfg.s3_bucket = store_parts['bucket']
+        cfg.s3_date_prefix = store_parts['date-prefix']
+        cfg.s3_path = store_parts['path']
```

Comment:
This will definitely work, but related to the configuration discussion, I'd rather we normalized that where possible to avoid needing this kind of code.